web3/main.py

```diff
@@ -125,3 +147,48 @@ def createBatch(self):
 
     def receive(self, requestid, timeout=0, keep=False):
         return self._requestManager.receive(requestid, timeout, keep)
+
+
+def hex_encode_abi_type(abi_type, value):
+    if is_array_type(abi_type):
+        sub_type = re.sub(r"\[[^]]*\]$", "", abi_type, 1)
+        return "".join([hex_encode_abi_type(sub_type, v) for v in value])
+    elif is_bool_type(abi_type):
+        return remove_0x_prefix(hex(value)).zfill(2)
+    elif is_uint_type(abi_type):
+        return positive_int_to_hex(value, size_of_type(abi_type))
+    elif is_int_type(abi_type):
+        bit_size = size_of_type(abi_type)
+        if value < 0:
+            return twos_compliment_of_negative(value, bit_size)
+        else:
+            return positive_int_to_hex(value, bit_size)
+    elif is_address_type(abi_type):
+        return remove_0x_prefix(value)
+    elif is_bytes_type(abi_type):
+        if is_0x_prefixed(value):
+            return remove_0x_prefix(value)
+        else:
+            return bytes_to_hex(value)
```

Comment:
I believe you'll run into the same issue as https://github.com/pipermerriam/web3.py/issues/215 . Try input `b'\x30\x78\xde\xad'`.

Perhaps you could assume that value of type `str` is hex-encoded, and value of type `bytes` is the raw bytes. Then you're hosed in Python 2.x, but maybe this could be a Python 3+ only feature (of course, I'm in favor of forcing the whole lib to py3).